system-test/datastore.ts

```diff
@@ -861,34 +859,32 @@ describe('Datastore', () => {
               },
             ]);
 
-            transaction.commit(err => {
+            transaction.commit(async err => {
               assert.ifError(err);
 
               // Incomplete key should have been given an ID.
               assert.strictEqual(incompleteKey.path.length, 2);
 
-              async.parallel(
-                [
-                  // The key queued for deletion should have been deleted.
-                  callback => {
-                    datastore.get(deleteKey, (err, entity) => {
-                      assert.ifError(err);
-                      assert.strictEqual(typeof entity, 'undefined');
-                      callback();
-                    });
-                  },
+              await Promise.all([
+                // Deletes the key that is in the deletion queue.
+                callback => {
```

Comment:
Mixing callback and promise based code is not good :)  These shouldn't be callbacks, and we should be using awaits throughout here.